app/services/analytics_events.rb

```diff
@@ -681,32 +681,224 @@ def idv_in_person_proofing_residential_address_submitted(**extra)
     track_event('IdV: in person proofing residential address submitted', **extra)
   end
 
-  def idv_in_person_proofing_address_submitted(**extra)
-    track_event('IdV: in person proofing address submitted', **extra)
+  # @param [String] flow_path
+  # @param [String] step
+  # @param [Integer] step_count
+  # @param [String] analytics_id
+  # @param [Boolean] irs_reproofing
+  # @param [Boolean] success
+  # @param [Hash] errors
+  # @param [Boolean] same_address_as_id
+  # address submitted by user
+  def idv_in_person_proofing_address_submitted(
```

Comment:
We should default the new parameters to `nil` unless we know they're explicitly passed in. Some of the FSM refactoring could also break these if we're not careful.

Reply:
triggering these events the only property that is sometimes nil is `same_address_as_id` ( in the state_id and address submission events) and this is depending on if DAV is on or off. Is your thought here that we want to make these all nillable so that when we remove the fsm files these logs won't break anything? I'd think we'd want to go back and update the events to make sure they're inline with what they should look like at that point

Reply:
Any of the events where the key may not be passed in - in either feature state - should have that parameter defaulted to `nil`. If the parameter is actually passed in as `nil` (but included as a key), then that won't be a problem. It's only if the key may not be present that we need to be concerned here.

If we suspect that FSM changes are the most likely ones to break this (and DAV & similar changes have been accounted for), then we could ensure that analytics stories/AC are present and prioritized in the epic to support compatibility with both before/after states for the FSM changes

Reply:
also as a quick note, if for some reason we didn't want to log keys in the eventual log for `nil` values, we can use `{}.compact` like we do elsewhere in this file

Reply:
ok u updated the params to have a default value of `nil` in case they don't get passed in. This have fixed some of the spec failures. There is however a failure as `same_address_as_id` is being flagged as pii.

Reply:
@svalexander I think you can set `pii_like_keypaths` to `[[: same_address_as_id]]` in order to fix that issue.

Reply:
it looks like we are doing that but it still fails the check https://github.com/18F/identity-idp/blob/b9c2334c551a99797de8f92a277b7df491ba2dde/app/services/idv/flows/in_person_flow.rb#L61